### PR TITLE
feat(dbt): add `get_asset_key_by_output_name`

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary & Motivation
Separating this out so that the initial example that's referenced in the RFC is working without needing to upgrade versions.

## How I Tested These Changes
local

### CODE REVIEW

**Comment by rexledesma** [comment]
Current dependencies on/for this PR:
* master
  * **PR #14790** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14790" target="_blank">Graphite</a> 
    * **PR #14791** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14791" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/14791?utm_source=stack-comment).

File: python_modules/libraries/dagster-dbt/dagster_dbt/cli/resources_v2.py [hunk 1]
```diff
@@ -126,9 +126,9 @@ def get_node_info_by_output_name(self, output_name: str) -> Mapping[str, Any]:
         """Get a dbt node's dictionary representation in the manifest by its Dagster output name."""
         return self.node_info_by_output_name[output_name]
 
-    def get_node_info_by_asset_key(self, asset_key: AssetKey) -> Mapping[str, Any]:
-        """Get a dbt node's dictionary representation in the manifest by its Dagster output name."""
-        return self.node_info_by_asset_key[asset_key]
+    def get_asset_key_by_output_name(self, output_name: str) -> AssetKey:
```
Comment (OwenKephart) [inline python_modules/libraries/dagster-dbt/dagster_dbt/cli/resources_v2.py:129]: In favor of this functionality, just want to raise that I think we should do a round of renaming the methods on this object, as right now they're a bit clunky. Can save that discussion for another PR, but roughly I'm thinking:

- node_info_to_{x} -> {x}_fn (e.g. `node_info_to_asset_key` -> `asset_key_fn`)
- `output_name` -> `output`
- get_{x}_by_{y} -> {x}_for_{y} (e.g. `get_asset_key_by_output_name` -> `asset_key_for_output_name`) (note [precedent](https://sourcegraph.com/github.com/dagster-io/dagster/-/blob/python_modules/dagster/dagster/_core/execution/context/compute.py?L315:9))
- `node_info` -> `dbt_info` (e.g. `get_node_info_by_output_name` -> `dbt_info_for_output`)

**Review by OwenKephart — APPROVED** [review]
note naming comment -- fine to resolve now or later

File: python_modules/libraries/dagster-dbt/dagster_dbt/cli/resources_v2.py (hunk 1 above)
Comment (rexledesma) [reply python_modules/libraries/dagster-dbt/dagster_dbt/cli/resources_v2.py:129]: Yeah we can do a mass rename before releasing `1.4`.

**Pull request merged**